src/main/java/com/mindee/parsing/common/ocr/OcrPage.java

```diff
@@ -0,0 +1,78 @@
+package com.mindee.parsing.common.ocr;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.mindee.geometry.PolygonUtils;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.stream.Collectors;
+import lombok.Getter;
+
+/**
+ * Represents a page.
+ */
+@Getter
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class OcrPage {
+
+  @JsonProperty("all_words")
+  private ArrayList<Word> words = new ArrayList<>();
+
+  /**
+   * Order all the words on the page into lines.
+   */
+  public ArrayList<ArrayList<Word>> toLines() {
```

Comment:
Does it make sense to only run toLines loops once and store the results in a field and return the field on future invocations?